### PR TITLE
Experimental Copy on Write

### PR DESCRIPTION
**Pull request by gregli-msft**

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/MutationScripts/CopyOnWrite.txt [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+// TABLE APPEND
+
+>> Set( aa, [1,2,3] )
+Table({Value:1},{Value:2},{Value:3})
+
+>> Set( ab, aa )
+Table({Value:1},{Value:2},{Value:3})
+
+>> Collect( ab, {Value:4} )
```
Comment (Shpakh-MSFT) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/MutationScripts/CopyOnWrite.txt:29]: Is this valid to mutate variable value like this?

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by gregli-msft** [comment]
Superseded by https://github.com/microsoft/Power-Fx/pull/1512

**Pull request closed without merging**